src/core/mesh/qgsmeshlayer.h

```diff
@@ -119,26 +122,48 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
     bool readXml( const QDomNode &layer_node, QgsReadWriteContext &context ) override;
     bool writeXml( QDomNode &layer_node, QDomDocument &doc, const QgsReadWriteContext &context ) const override;
 
-    //! Return the provider type for this layer
+    //! Returns the provider type for this layer
     QString providerType() const;
 
-    //! return native mesh (nullprt before rendering)
+    //! Returns native mesh (nullprt before rendering)
     QgsMesh *nativeMesh() SIP_SKIP;
 
-    //! return triangular mesh (nullprt before rendering)
+    //! Returns triangular mesh (nullprt before rendering)
     QgsTriangularMesh *triangularMesh() SIP_SKIP;
 
-    //! Returns a line symbol used for rendering native mesh.
-    QgsSymbol *nativeMeshSymbol();
+    //! Returns rendrer settings
+    QgsMeshRendererMeshSettings rendererNativeMeshSettings() const SIP_FACTORY;
 
-    /**
-     * Returns a line symbol used for rendering of triangular (derived) mesh.
-     * \see toggleTriangularMeshRendering
-     */
-    QgsSymbol *triangularMeshSymbol();
+    //! Sets new rendering settings, triggers repaint
+    void setRendererNativeMeshSettings( const QgsMeshRendererMeshSettings &settings );
+
+    //! Returns rendrer settings
+    QgsMeshRendererMeshSettings rendererTriangularMeshSettings() const SIP_FACTORY;
```

Comment:
No need to mark these methods as SIP_FACTORY - we are returning by value, not a new pointer